### PR TITLE
fixes gh-1207 add box.error.set and update box.error.new functions

### PR DESCRIPTION
**Pull request by Onvember**

### CODE REVIEW

File: doc/reference/reference_lua/box_error.rst [hunk 1]
```diff
@@ -149,10 +152,11 @@ Below is a list of all ``box.error`` functions.
 
 .. function:: box.error.new(code, errtext [, errtext ...])
 
-    Create an error object, but do not throw.
+    Create an error object, but doesn't set it to Tarantool's diagnostic area as
+    :ref:`box.error() <box_error-error>` does.
     This is useful when error information should be saved for later retrieval.
-    The parameters are the same as for :ref:`box.error() <box_error-error>`,
-    see the description there.
+    To set error in diagnostic area explicitly use
```
Comment (lenkis) [inline doc/reference/reference_lua/box_error.rst:158]:
```suggestion
    To set an error in a diagnostic area explicitly use 
```

File: doc/reference/reference_lua/box_error.rst [hunk 2]
```diff
@@ -173,6 +177,32 @@ Below is a list of all ``box.error`` functions.
...
         ...
+        tarantool> box.error.last()
+        ---
+        - nil
 
+.. _box_error-set:
 
+.. function:: box.error.set(error object)
 
+    Set an error in diagnostic area explicitly. Accepts error object and set it
```
Comment (lenkis) [inline doc/reference/reference_lua/box_error.rst:188]:
```suggestion
    Set an error in a diagnostic area explicitly. Accepts an error object and sets it
```

File: doc/reference/reference_lua/box_error.rst [hunk 3]
```diff
@@ -173,6 +177,32 @@ Below is a list of all ``box.error`` functions.
...
+    as the last system error (i.e. becomes available via ``box.error.last()``).
```
Comment (lenkis) [inline doc/reference/reference_lua/box_error.rst:189]:
```suggestion
    as the last system error (i.e. becomes available via :ref:`box.error.last() <box_error-last>`).
```

File: doc/reference/reference_lua/box_error.rst [hunk 4]
```diff
@@ -173,6 +177,32 @@ Below is a list of all ``box.error`` functions.
...
+
+    **Example:**
```
Comment (lenkis) [inline doc/reference/reference_lua/box_error.rst:191]: I have different output here, please double-check with the dev team.

Tarantool 2.4.1-6-g83c0356a1
Ubuntu 18.04

```
tarantool> box.cfg()
tarantool> err = box.error.new({code = 111, reason = "cause"})
---
...
tarantool> tarantool> box.error.last()
---
- error: '[string "return tarantool> box.error.last()"]:1: attempt to compare two
    nil values'
...
tarantool> box.error.set(err)
---
...

tarantool> box.error.last()
---
- cause
...
```

Comment (Onvember) [reply doc/reference/reference_lua/box_error.rst:191]: fixed in f614b5d

File: locale/ru/LC_MESSAGES/reference/reference_lua/box_error.po [hunk 5]
```diff
@@ -255,16 +241,18 @@ msgstr ""
...
+"<box_error-error>` does. This is useful when error information should be "
+"saved for later retrieval. To set an error as the last explicitly use "
+":ref:`box.error.set() <box_error-set>`."
 msgstr ""
-"Создание ошибки без выдачи. Используется, когда необходимо сохранить "
-"информацию об ошибке для последующей выборки. Используются такие же "
-"параметры, как в :ref:`box.error() <box_error-error>`, см. описание по "
-"ссылке."
+"Создает объект ошибки, но не отправляет в область диагностики, как это "
+"деалет :ref:`box.error() <box_error-error>`. Это полезно, когда "
```
Comment (lenkis) [inline locale/ru/LC_MESSAGES/reference/reference_lua/box_error.po:252]:
```suggestion
"делает :ref:`box.error() <box_error-error>`. Это полезно, когда "
```

**Pull request merged**